ckb-transaction-cobuild/src/lib.rs

```diff
@@ -169,124 +169,151 @@ pub fn parse_message() -> Result<([u8; 32], Vec<u8>), Error> {
-                                hasher.update(output_cell_data.as_slice());
-                            }
-                            self.output_cell_counter += output_cells as usize;
-
-                            // otx cell deps
-                            hasher.update(&cell_deps.to_le_bytes());
-                            let cell_dep_iter =
-                                raw_tx.cell_deps().into_iter().skip(self.cell_deps_counter);
-                            for cell_dep in cell_dep_iter.take(cell_deps as usize) {
-                                hasher.update(cell_dep.as_slice());
-                            }
-                            self.cell_deps_counter += cell_deps as usize;
-
-                            // otx header deps
-                            hasher.update(&header_deps.to_le_bytes());
-                            let header_dep_iter = raw_tx
-                                .header_deps()
-                                .into_iter()
-                                .skip(self.header_deps_counter);
-                            for header_dep in header_dep_iter.take(header_deps as usize) {
-                                hasher.update(header_dep.as_slice());
-                            }
-                            self.header_deps_counter += header_deps as usize;
-
-                            let mut result = [0u8; 32];
-                            hasher.finalize(&mut result);
-                            return Some((result, otx.seals().to_entity()));
-                        } else {
-                            self.input_cell_counter += input_cells as usize;
-                            self.output_cell_counter += output_cells as usize;
-                            self.cell_deps_counter += cell_deps as usize;
-                            self.header_deps_counter += header_deps as usize;
-                        }
-                    }
-                    _ => return None,
+        for otx_structure in &mut self.inner_iter {
+            let otx = otx_structure.otx.as_reader();
+            let input_cells: u32 = otx.input_cells().unpack();
+            let output_cells: u32 = otx.output_cells().unpack();
+            let cell_deps: u32 = otx.cell_deps().unpack();
+            let header_deps: u32 = otx.header_deps().unpack();
+            let mut input_lock_hash_iter = QueryIter::new(load_cell_lock_hash, Source::Input)
+                .skip(otx_structure.input_cell_start)
+                .take(input_cells as usize);
+            if input_lock_hash_iter.any(|lock_hash| lock_hash == self.current_script_hash) {
+                let mut hasher = new_otx_blake2b();
+                // message
+                hasher.update(otx.message().as_slice());
+
+                // otx inputs
+                hasher.update(&input_cells.to_le_bytes());
+                let input_iter = raw_tx
+                    .inputs()
+                    .into_iter()
+                    .skip(otx_structure.input_cell_start)
```

Comment:
The `skip(n)` method doesn't report any error when the count of iterator smaller than`n`.
The same to`take` method.

Reply:
2 points:

* Note that current code already follows this behavior, I'm just moving code around here, I don't really change anything
* If you think about it: is it really a problem: when one has less input cells than the OTX needs, a different seal will be calculated, making the signature validation failed. So to me, I don't think an error is required here.